python/idsse_common/test/test_validate_schema.py

```diff
@@ -129,8 +206,9 @@ def criteria_message() -> dict:
                 "combined": "A AND B",
             },
         ],
-        "data": {
-            "A": {
+        "parts": [
+            {
+                "name": "A",
```

Comment:
What was the motivation for changing the parts/sub-conditions data structure from a dictionary (JSON object) to a list (JSON array)?

I think either is fine. With a dictionary, we'd look up a part named "A" with
```python
data.A
```
vs with an array we'd loop through the array with something like
```python
[part for part in parts if part.name == "A"]
```

Reply:
I had two reasons for making the switch, hopefully they offset the less direct retrieval. First I didn't like the fact the keys (above you have "A", but there are three, conditionKey, locationKey, and productKey) were referenced without identifying them. Is the top key the condition or location? I like that with the current structure it is more human readable. The second reason was that it is really hard to write a schema for the previous tree structure with the triple nested keys. I have one more reason, the new struct follows the pattern used in the criteria schema (and thus in the rest of the eventPort), for example the condition is a list of objects and not a tree structure.

Reply:
After some quick reading, I agree that direct lookup with a dictionary (sometimes called "hash table") is just not worth it.

They may be slightly less code to retrieve, and in theory faster lookup time, but I think the difference is negligible for how big the list of condition parts will be. And human readability is a bigger deal in my mind.